File: gradle-multi-modules/module-web/src/main/java/com/example/api/MemberController.java
package com.example.api;

import com.example.domain.Member;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class MemberController {

    @GetMapping("/")
    public Member get() {
        return new Member("jojoldu", "jojoldu@gmail.com");
    }
}

File: Kotlin/spring-project/src/main/kotlin/com/example/demo/controller/MemberController.kt
package com.example.demo.controller

import com.example.demo.service.MemberService
import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/members")
class MemberController(
        private val memberService: MemberService
) {

    @PostMapping
    fun create(@RequestBody memberCreateDto: MemberCreateDto): ResponseEntity<MemberResponseDto> {

        val memberResponseDto = memberService.save(memberCreateDto)
        val uri = linkTo(MemberController::class.java).slash(memberResponseDto.id).toUri()
        return ResponseEntity.created(uri).body(memberResponseDto)
    }
}